### ISSUE TITLE
Comments do not work if Image Added (my first error posting)

### ISSUE THREAD

**Issue by datamd** [ClosedXML/ClosedXML#712]
## Read and complete the full issue template
(My First Error Post) There is error I have if I add a comment and if I have also added an image ? 
The error only happens on opening a generated excel file "we found a problem with some content in 'filename.xlsx'. Do you want us to try to recover as much as we can? if you trust the source of this workbook, click yes"

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature

**Version of ClosedXML**

e.g. 0.87.1

**What is the current behavior?**
The resulting file is currupt with error message "we found a problem with some content in 'filename.xlsx'. Do you want us to try to recover as much as we can? if you trust the source of this workbook, click yes"
Complete this.

**What is the expected behavior or new feature?**
Both Add Comments and Add Image should both work together 
Complete this.

**Did this work in previous versions of our tool?  Which versions?**
I have only used this latest version
Yes/No/v0.XX

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**
I only get an error is I call this line of code 
AddImage(wb, wbSummary.Name, 1, 1, @"Q:\images\Logo.jpg");

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
public void Main()
{
    // Code standards:
    // - Fully runnable. I should be able to copy and paste this code into a 
    //   console application and run it without having to edit it much.
    // - Declare all your variables (this follows from the previous point)
    // - The code should be a minimal code sample to illustrate issue. The code 
    //   samples on the wiki are good examples of the terseness that I want. Don't
    //   post your full application.
}
```
- [ ] I attached a sample spreadsheet.  (You can drag files on to this issue)

**Comment by datamd** [comment]
I hope my first minimal comment is just about good enough ?

**Comment by igitur** [comment]
Welcome, but sorry, I require a full code sample. I'm squashing bugs in. My coffee breaks at my actual job, so you need to make it easy for me.

**Comment by vbjay** [comment]
You need to learn the concept of an mvce. How can we run your code when we
don't have your DB. You need to make a sample that we can put in a console
project or linqpad and just run.
https://stackoverflow.com/help/mcve

**Comment by igitur** [comment]
# Code guidelines 

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
public void Main()
{
    // Code standards:
    // - Fully runnable. I should be able to copy and paste this code into a 
    //   console application and run it without having to edit it much.
    // - Declare all your variables (this follows from the previous point)
    // - The code should be a minimal code sample to illustrate issue. The code 
    //   samples on the wiki are good examples of the terseness that I want. Don't
    //   post your full application.
}
```

**Comment by datamd** [comment]
(reply all) 

OK I did a minimal cut down version (MCVE) below and it works fine ... 

using ClosedXML.Excel;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClosedXML_Issue_172
{
    class Program
    {
        static void Main(string[] args)


        {
            using (XLWorkbook wb = new XLWorkbook())
            {
                var wbSummary = wb.AddWorksheet("Summary");
                wbSummary.Cell("D4").Value = "Test Value";

                AddImage(wb, wbSummary.Name, 1, 1, "image.jpg");

                // Set the comment as visible
                wbSummary.Cell("D4").Comment.SetVisible().AddText("This is a comment");

                wb.SaveAs("outputfile_stage2.xlsx");
                Console.WriteLine("Finished ... this minimal sample works fine ");
            }


            void AddImage(XLWorkbook wb, string sheetName, int col, int row, string strImageLocation)
            {
                if (!File.Exists(strImageLocation)) return;
                var ws = wb.Worksheet(sheetName);
                var image = ws.AddPicture(strImageLocation);
                image.MoveTo(ws.Cell(row, col).Address);
                image.Scale(.5); // optional: resize picture
                wb.SaveAs("outputfile_stage1.xlsx");
                // wb.Save(); // System.InvalidOperationException: 'This is a new file. Please use one of the 'SaveAs' methods.'

            }

        }
    }
}

**Comment by igitur** [comment]
If I run your code, I get this file:
![image](https://user-images.githubusercontent.com/145854/37024729-581df2be-2132-11e8-9114-d8e68c66f3fc.png)
[outputfile_stage2.xlsx](https://github.com/ClosedXML/ClosedXML/files/1784315/outputfile_stage2.xlsx)

Please ensure you're using the latest version of ClosedXML.

**Comment by datamd** [comment]
Inside AddImage if you remove the wb.save() then you get the error trying to open the file in excel ... this was the issue which I was getting ... so fixed my doing a wb.save() which I hope does not slow things down too much ?

**Comment by igitur** [comment]
` wb.Save(); ` is already commented out. Removing it won't make a difference. 

Please edit the code above so that it actually replicates your problem.

**Comment by datamd** [comment]
What I was finding was that if Inside AddImage if you comment out both .Save and .SaveAs on a NEW sheet   it processes fine but the resulting excel file when opening comes up with an error message

**Comment by igitur** [comment]
**Please edit the code above so that it actually replicates your problem.**

**Comment by datamd** [comment]
The following create me a excel file I cannot load up in excel 







using ClosedXML.Excel;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClosedXML_Issue_172
{
    class Program
    {
        static void Main(string[] args)


        {
            using (XLWorkbook wb = new XLWorkbook())
            {
                var wbSummary = wb.AddWorksheet("Summary");
                wbSummary.Cell("D4").Value = "Test Value";

                AddImage(wb, wbSummary.Name, 1, 1, "image.jpg");

                // Set the comment as visible
                wbSummary.Cell("D4").Comment.SetVisible().AddText("This is a comment");

                wb.SaveAs("outputfile_stage2.xlsx");
                Console.WriteLine("Finished ... this causes me to have a .xlsx file I cannot open in excel");
            }


            void AddImage(XLWorkbook wb, string sheetName, int col, int row, string strImageLocation)
            {
                if (!File.Exists(strImageLocation)) return;
                var ws = wb.Worksheet(sheetName);
                var image = ws.AddPicture(strImageLocation);
                image.MoveTo(ws.Cell(row, col).Address);
                image.Scale(.5); // optional: resize picture
                //wb.SaveAs("outputfile_stage1.xlsx"); // commented out because I would only like to save to disk once 
                // wb.Save(); // I cannot do a .Save because with a New Sheet I get "System.InvalidOperationException: 'This is a new file. Please use one of the 'SaveAs' methods.'"

            }

        }
    }
}

**Comment by igitur** [comment]
Ok, I can replicate the issue now. It's not related to the comments. I'll submit a fix.

**Comment by igitur** [comment]
Sorry, comments are treated internally as "legacy drawings", so comments are related to this bug.

**Comment by igitur** [comment]
Pleas test the build at https://ci.appveyor.com/project/Pyropace/closedxml/build/0.92.854/artifacts and let me know if it fixes the problem.

**Comment by datamd** [comment]
Hi Francois

No : - (

I copied across and replaced the 2 files and did a rebuild and it still has
the same problem

Attached  is *outputfile_stage2.xlsx*

Whick may help you in dubugging perhaps ?

Regards
Mark Dicken

**Comment by igitur** [comment]
Hmm, if you reply via email, attachments aren't saved to this issue. Rather use the web interface.

**Comment by datamd** [comment]
[outputfile_stage2.xlsx](https://github.com/ClosedXML/ClosedXML/files/1785192/outputfile_stage2.xlsx)

file attached ... hope this help you debugging

**Comment by igitur** [comment]
Sorry, I don't think you're referencing the new build correctly. Please delete all instances of ClosedXML locally and download and reference from https://ci.appveyor.com/project/Pyropace/closedxml/build/0.92.854/artifacts again.

**Comment by datamd** [comment]
PERFECT ... that works : - )

GREAT

I can now add Images and Comments with just one SaveAs at the very end ...
cheers

This GITHUB I think is great ...

Regards
Mark

**Comment by datamd** [comment]
Issue Closed

**Comment by igitur** [comment]
Ok, that build that I gave you is just for a temporary solution. The fix will be included in the 0.92 release.
